hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/BucketAssignFunction.java

```diff
@@ -189,15 +195,17 @@ public void processElement(I value, Context ctx, Collector<O> out) throws Except
 
     // The dataset may be huge, thus the processing would block for long,
     // disabled by default.
-    if (bootstrapIndex && !partitionLoadState.contains(partitionPath)) {
+    if (bootstrapIndex && !Boolean.TRUE.equals(stateLoadFlag.value())) {
```

Comment:
The default value of `bootstrapIndex` will cause wrong operation type(UPDATE -> INSERT) if the table is not a newly created